llms/mlx_lm/utils.py

```diff
@@ -98,11 +99,14 @@ def generate_step(
     """
 
     def sample(logits: mx.array) -> Tuple[mx.array, float]:
+        softmax_logits = mx.softmax(logits / temp if temp > 0 else logits)
```

Comment:
@awni  This fixes the error `UnboundLocalError: local variable 'prob' referenced before assignment`.

Reply:
Oops, I did test it carefully enough. I think we may need unit tests for this package if it keeps expandinf

Reply:
Also I think this is how it was before, but I wonder about the logic of returning the plain softmax if `temp == 0` but the temp version otherwise. It seems strange that `temp = 1e-4` will give you much more confident probabilities than `temp = 0` (which is the most confident). It seems more reasonable to me to always ignore the temperature in the probability computation.

Reply:
My understanding was that this was meant to show how the token is selected. My intuition was that whatever token is selected, we return its probability. but I didn't really understand why `temp = 1e-4` would give us the most confident probability. :(

Reply:
If you use a really small temperature the max probability gets closer and closer to 1. As temp -> 0 , the max probability is exactly 1 (aka argmax sampling). Hence the inconsistency. Yea I would prefer to just set `prob = mx.softmax(logits)[token]` in all cases as I think it is more consistent / decoupled from how you sample.

Reply:
Would you mind making that change and then we can merge?

Reply:
Yeah, that makes sense. Sorry, I misread the `1e-4` as `1e4`. I will update it now.